src/object/cli_shard.c

```diff
@@ -390,12 +391,14 @@ obj_ec_iom_merge(struct obj_reasb_req *reasb_req, uint32_t shard,
 	/* merge iom_recx_lo */
 	lo = src->iom_recx_lo;
 	end = DAOS_RECX_END(lo);
-	lo.rx_nr = min(end, roundup(lo.rx_idx + 1, cell_rec_nr)) - lo.rx_idx;
-	if (shard < obj_ec_data_tgt_nr(oca))
+	if (end > 0 && is_data_shard) {
```

Comment:
lo is also 0 for parity shard? no?

Reply:
Btw: this to and hi merging seems a bit duplicate?

Reply:
could be merge to one func with a flag to differentiate high/low, seems not quite necessary although.